website/search/elastic_search.py

```diff
@@ -582,10 +683,11 @@ def update_file(file_, index=None, delete=False):
     file_guid = file_.get_guid(create=False)
     if file_guid:
         guid_url = '/{file_guid}/'.format(file_guid=file_guid._id)
+    # File URL's not provided for preprint files, because the File Detail Page is blocked
     file_doc = {
         'id': file_._id,
-        'deep_url': file_deep_url,
-        'guid_url': guid_url,
+        'deep_url': None if isinstance(target, Preprint) else file_deep_url,
+        'guid_url': None if isinstance(target, Preprint) else guid_url,
```

Comment:
Since we are blocking Preprint file detail page, return none if this is a preprint file. On search UI, if no url exists, then we won't provide a link to the preprint.